### PR TITLE
Create laws and tests: optical distance, optical distance difference

### PR DESCRIPTION
**Pull request by Voldoned**

### CODE REVIEW

File: symplyphysics/laws/optics/optical_distance_from_geometric_distance_and_refractive_index.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from sympy import (Eq, solve)
+from symplyphysics import (units, Quantity, Symbol, print_expression, dimensionless, validate_input,
+    validate_output)
+
+# Description
+## Optical distance L is a geometric distance l in optical environment with refractive index n
+
+# Law: L = n * l
```
Comment (blackyblack) [inline symplyphysics/laws/optics/optical_distance_from_geometric_distance_and_refractive_index.py:8]: We already have this: https://github.com/blackyblack/symplyphysics/blob/master/symplyphysics/laws/waves/optical_path_length_from_geometric_path_length_and_refractive_index.py

File: symplyphysics/laws/optics/optical_distance_difference_from_optical_distances.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from sympy import Eq, solve
+
+from symplyphysics import Symbol, units, print_expression, Quantity, \
+    validate_input, validate_output
+
+# Description
+## The optical difference in the course of two rays is the difference
+## in the optical distances traversed by each of the rays
+
+# Law: delta = L2 - L1
```
Comment (blackyblack) [inline symplyphysics/laws/optics/optical_distance_difference_from_optical_distances.py:10]: Do we really need this law? We do not have laws for force difference, energy difference, or whatever difference. What's so special about optical distance difference?

Comment (Voldoned) [reply symplyphysics/laws/optics/optical_distance_difference_from_optical_distances.py:10]: I created this definition for possible clarification of some points in the derivation of other formulas for the difference in optical distance (For example, Young's experiment, thin wedge, Newton's rings)

This material on spatial coherence writes the difference in optical distances using this definition (both beams are in the air, so the refractive indices there are equal to 1): https://studme.org/341473/matematika_himiya_fizik/vliyanie_razmerov_istochnika_sveta_prostranstvennaya_kogerentnost

File: symplyphysics/laws/optics/optical_distance_difference_from_optical_distances.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Where:
+# L1 - optical distance for 1st wave
+# L2 - optical distance for 2nd wave
+
+optical_distance_wave_1 = Symbol("", units.length)
```
Comment (blackyblack) [inline symplyphysics/laws/optics/optical_distance_difference_from_optical_distances.py:15]:
```suggestion
optical_distance1 = Symbol("", units.length)
```

Comment (blackyblack) [inline symplyphysics/laws/optics/optical_distance_difference_from_optical_distances.py:15]: Where is the name?

**Review by blackyblack — APPROVED** [review]

**Pull request merged**